SOP for miRNA annotation
========================

# This is a pipeline for miRNA annotation

## Software requirement for this annotation pipeline

* Trimmomatic v0.36
* Anaconda2/4.2.0 
* SHRiMP v2.2.3  
* R v3.4.3 

### Resources

SHRiMP is a software package for aligning genomic reads against a target genome. 
It was primarily developed with the multitudinous short reads of next generation sequencing machines in mind, 
as well as Applied Biosystem's colourspace genomic representation. For more infor please check: http://compbio.cs.toronto.edu/shrimp/ and https://github.com/compbio-UofT/shrimp

For remove the adapter, you can find the detailed guideline as below: http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf
with the following link, you will find the mirbase database, after download the mirbase database, you can retrieve the species what you are working on if you want. 

```
wget ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.gz   
wget ftp://mirbase.org/pub/mirbase/CURRENT/mature.fa.zip    
wget ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.zip
wget ftp://mirbase.org/pub/mirbase/CURRENT/genomes/hsa.gff3 
wget ftp://mirbase.org/pub/mirbase/CURRENT/miFam.dat.zip

```
## Generating hairpin.human.fa and mature.human.fa reference file for miRNA annotation with miRbase.sh

We project the database with:
$SHRIMP_FOLDER/utils/project-db.py --seed 00111111001111111100,00111111110011111100,00111111111100111100,00111111111111001100,00111111111111110000 \
--h-flag --shrimp-mode ls /home/mxie/Projects/miRNA/analysis/miRbase1/hairpin.human.fa
$SHRIMP_FOLDER/utils/project-db.py --seed 00111111001111111100,00111111110011111100,00111111111100111100,00111111111111001100,00111111111111110000 \
--h-flag --shrimp-mode ls /home/mxie/Projects/miRNA/analysis/miRbase1/mature.human.fa

Here, we  used 5  seeds of weight 14 and span 20 which  allow for  2 initial
mismatches (the leading 00), followed by a match of length 7bp (the 6 following
1s), followed by another match of various forms.

Note:  The seeds  are  applied to  the color space  translations of both  the
database and the  reads. Hence, a  string of 6  consecutive 1s corresponds  to 6
consecutive matching colours, which are in particular generated by a string of 7
consecutive matching letters.

```
nano miRbase.sh
#!/bin/sh
set -euo pipefail
wget ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.gz   
wget ftp://mirbase.org/pub/mirbase/CURRENT/mature.fa.zip    
wget ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.zip
wget ftp://mirbase.org/pub/mirbase/CURRENT/genomes/hsa.gff3 
wget ftp://mirbase.org/pub/mirbase/CURRENT/miFam.dat.zip
gunzip hairpin.fa.gz
unzip mature.fa.zip
grep sapiens mature.fa |wc   # 2588 
grep sapiens hairpin.fa |wc  # 1881 
# Homo sapiens
perl -alne '{if(/^>/){if(/Homo/){$tmp=1}else{$tmp=0}};next if $tmp!=1;s/U/T/g if !/>/;print }' hairpin.fa >hairpin.human.fa
perl -alne '{if(/^>/){if(/Homo/){$tmp=1}else{$tmp=0}};next if $tmp!=1;s/U/T/g if !/>/;print }' mature.fa >mature.human.fa
cd /sw/hgcc/Pkgs/SHRiMP/2.2.3
export SHRIMP_FOLDER=$PWD
cd /home/mxie/Projects/miRNA/analysis/miRbase1
$SHRIMP_FOLDER/utils/project-db.py --seed 00111111001111111100,00111111110011111100,00111111111100111100,00111111111111001100,00111111111111110000  --h-flag --shrimp-mode ls /home/mxie/Projects/miRNA/analysis/miRbase1/hairpin.human.fa
$SHRIMP_FOLDER/utils/project-db.py --seed 00111111001111111100,00111111110011111100,00111111111100111100,00111111111111001100,00111111111111110000  --h-flag --shrimp-mode ls /home/mxie/Projects/miRNA/analysis/miRbase1/mature.human.fa

```

## Parse file id and merge multiple Fastq files into one big Fastq file


```
nano parseid.py
import os
import csv
os.chdir("/home/mxie/Projects/miRNA/analysis/Past86samples")
filelist=[]
for f in os.listdir():
    f_name,f_ext = os.path.splitext(f)
    f1 = f_name.split('_')[0]
    f2 = f_name.split('_')[1]
    f3 = f_name.split('_')[2]
    newname= "_".join([f1,f2,f3])
    filelist.append(newname)
cleanlist = []
[cleanlist.append(x) for x in filelist if x not in cleanlist]
with open("/home/mxie/Projects/miRNA/analysis/pastsample/cleanlist.csv",'w') as resultFile:
    wr = csv.writer(resultFile, dialect='excel')
    wr.writerow(cleanlist)
with open ("/home/mxie/Projects/miRNA/analysis/pastsample/cleanlist.txt","w")as file:
   file.write('\n'.join(cleanlist))
```
```   
nano mergefastqfile.sh
#!/bin/sh
set -euo pipefail   
for file in $(cat /home/mxie/Projects/miRNA/analysis/pastsample/cleanlist.txt);
do
zcat $file* | pigz > /home/mxie/Projects/miRNA/analysis/pastsample/$file.fastq.gz; 
done   
   
qsub -cwd -pe smp 24 mergefastqfile.sh

```

## Remove the adapter with removeadaptor.sh by using trimmomatic

please note: For each step, if the directory is not specified, bassically, it is in current directory. You may can find out the directory with the output from the past step
For example:
In previous step, we are in /home/mxie/Projects/miRNA/analysis/pastsample/
The output will be in /home/mxie/Projects/miRNA/analysis/pastsample/trimmed, before this implement, please make sure create the trimmed directory  

``` 
nano removeadaptor.sh
#!/bin/sh
set -euo pipefail
ls *fastq.gz |while read id
do
echo $id
java -Xmx2g -jar /sw/hgcc/Pkgs/Trimmomatic/0.36/trimmomatic-0.36.jar SE -phred33 $id  ./trimmed/${id%%.*}_miRNAtrimmed.fastq.gz ILLUMINACLIP:/home/mxie/Projects/miRNA/analysis/illumina_RGA_3p_adapter.fa:2:30:10;
done   
qsub -cwd -pe smp 24 removeadaptor.sh

``` 

## Do quality control and show the length distribution with lengthdistribution.sh 

please note: For each step, if the directory is not specified, bassically, it is in current directory. You may can find out the directory with the output from the past step
For example:
In previous step, we are in /home/mxie/Projects/miRNA/analysis/pastsample/trimmed

```
nano lengthdistribution.sh
#!/bin/sh
set -euo pipefail
ls *_miRNAtrimmed.fastq.gz |while read id
do
echo $id
gunzip -c $id | awk '{if(NR%4==2) print length($1)}' | sort -n | uniq -c >${id%%.*}_length.txt;
done
qsub -cwd -pe smp 24 lengthdistribution.sh

```
Please note: you may need change setwd("D:/Emoryjob/miRNA/trimmedlength/") to your personal directory

```
setwd("D:/Emoryjob/miRNA/trimmedlength/")

file_list <- list.files()
# Create Line Chart
# get the range for the x and y axis 
xrange <- range(1,105) 
yrange <- range(1,1.8e+07) 
# set up the plot 
plot(xrange, yrange, type="n", xlab="Readlength",
     ylab="Countnumber" ) 
# add lines 
for (i in 1:(length(file_list)-1)) { 
  lenData <- read.table(file_list[i], sep ='')
  names(lenData) <- c("Countnumber", "Readlength")
  lenData<- as.data.frame(lenData)
  lines(lenData$Readlength, lenData$Countnumber, type="b", col="blue") 
} 
# add a title and subtitle
title("Read Length Distribution", "86 samples of line plot")

```

## Mapping miRNA to miRbase reference with mapping.sh

We map the reads with:

$SHRIMP_FOLDER/bin/gmapper-ls --strata -o 1 --qv-offset 33 -Q -L /home/mxie/Projects/miRNA/analysis/miRbase/mature.human-ls $id \
-N 14 --mode mirna -w 170% -E --un ${id}_unmapped-miRNA.fastq \
>${id%%.*}_mapping.sam 2>${id%%.*}_maping.log;

The options used above have the following meaning:
-L /home/mxie/Projects/miRNA/analysis/miRbase/mature.human-ls : Load genome projection.
-N 14 : Use 14 threads of control.
-w 170% : Enlarge the length  of the genome window against  which each read is being mapped.
"--mode mirna" is equivalent with:
loading the 5 seeds mentioned above; plus
"-H -n 1 -w 100% -U -a 0 -g -255 -q -255 -Z"  
[ -E/--sam ]
Select SAM output format. (This is on by default as of v2.3.0.)

If you want to look into it more detail, please check with: https://github.com/compbio-UofT/shrimp

```
nano mapping.sh
#!/bin/sh
set -euo pipefail
cd /sw/hgcc/Pkgs/SHRiMP/2.2.3
export SHRIMP_FOLDER=$PWD
cd /home/mxie/Projects/miRNA/analysis/pastsample/trimmed
ls *miRNAtrimmed.fastq.gz |while read id
do
echo $id
$SHRIMP_FOLDER/bin/gmapper-ls --strata -o 1 --qv-offset 33 -Q -L /home/mxie/Projects/miRNA/analysis/miRbase/mature.human-ls $id \
-N 14 --mode mirna -w 170% -E --un ${id}_unmapped-miRNA.fastq \
 >${id%%.*}_mapping.sam 2>${id%%.*}_maping.log;
done

qsub -cwd -pe smp 24 mapping.sh

```

## Counting the reads mapped for each miRNA

The idea behind here is to count the miRNA raw number, sum up the miRNA raw number with for loop in perl and save the sum up number into file
please note: For each step, if the directory is not specified, bassically, it is in current directory. You may can find out the directory with the output from the past step

```
nano counting.sh
#!/bin/sh
set -euo pipefail
ls  *.sam| while read id
do
echo $id
/sw/hgcc/Pkgs/samtools/1.5/bin/samtools view  -SF 4 $id |perl -alne '{$h{$F[2]}++}END{print "$_\t$h{$_}" foreach sort keys %h }'  > ${id%%.*}.mature.counts;
done
qsub -cwd -pe smp 24 counting.sh

```

## Do quality control and show the mapped length distribution with mappinglenthdistribution.sh

The idea behind here is to calculate the number of mapped reads with different length and make plot to show the distribution of mapped reads
please note: For each step, if the directory is not specified, bassically, it is in current directory. You may can find out the directory with the output from the past step

```
nano mappinglenthdistribution.sh
#!/bin/sh
set -euo pipefail
ls  *.sam| while read id
do
echo $id
/sw/hgcc/Pkgs/samtools/1.5/bin/samtools view $id | gawk '{mlarr[length($10)]++}END{for (len = 1; len <=101; len++) printf("%d\t%d\n",len,mlarr[len])}' | sort -k 1n,1n  > ${id%%.*}_mapped_length_distribution_data.txt
done
qsub -cwd -pe smp 24 mappinglenthdistribution.sh

```
Please note: you may need change setwd("D:/Emoryjob/miRNA/mappedlength/") to your personal directory. You may can find out the directory with the output from the past step

```
setwd("D:/Emoryjob/miRNA/mappedlength/")
file_list <- list.files()
# Create Line Chart
# get the range for the x and y axis 
xrange <- range(1,105) 
yrange <- range(1,1.8e+07) 
# set up the plot 
plot(xrange, yrange, type="n", xlab="Mapped length",
     ylab="Countnumber" ) 
for (i in 1:(length(file_list)-1)){
  lenData <- read.table(file_list[i], sep ='')
  names(lenData) <- c("mappinglength", "count")
  lenData<- as.data.frame(lenData)
  lines(lenData$mappinglength,lenData$count, type = 'b', col="blue")
}
# add a title and subtitle
title("Mapped Length Distribution", "86 samples of line plot")

```



